src/view/com/post-thread/PostThread.tsx

```diff
@@ -215,24 +256,53 @@ function PostThreadLoaded({
           offset: pageY,
         })
       }
-      if (isNative) {
-        highlightedPostRef.current?.measure(
-          (_x, _y, _width, _height, _pageX, pageY) => {
-            onMeasure(pageY)
-          },
-        )
-      } else {
-        // Measure synchronously to avoid a layout jump.
-        const domNode = highlightedPostRef.current
-        if (domNode) {
-          const pageY = (domNode as any as Element).getBoundingClientRect().top
-          onMeasure(pageY)
-        }
+      // Measure synchronously to avoid a layout jump.
+      const domNode = highlightedPostRef.current
+      if (domNode) {
+        // @ts-ignore web only
+        const pageY = (domNode as any as Element).getBoundingClientRect().top
+        onMeasure(pageY)
       }
       needsScrollAdjustment.current = false
     }
   }, [thread])
 
+  // HACK
+  // Until we can get maintainVisibleContentPosition to work regardless of how many posts we have rendered at the
+  // top, this will simulate a "load" when we reach the top, to allow for the next items to render without jumping
+  const onStartReached = React.useCallback(() => {
+    // Get the first post. We need to get the second item in the array if the first one is the TOP_COMPONENT
+    let first = posts?.[0]
+    if (first === TOP_COMPONENT) {
+      first = posts?.[1]
+    }
+
+    // We do nothing in these situations
+    // - We are already prepending
+    // - This is not a thread post
+    // - There are no more parents to show
+    // - The top post is the highlighted post
+    if (
+      isPrepending.current ||
+      !isThreadPost(first) ||
+      !first.parent ||
+      first.ctx.isHighlightedPost
+    ) {
+      return
+    }
+
+    // Set the ref to true for the onMomentumScrollEnd callback
+    isPrepending.current = true
+  }, [posts])
+
+  // We want the scroll animation to settle before we increment the page count. We will check every time the scroll
+  // animation ends and prepend posts once settled if necessary.
+  const onMomentumScrollEnd = React.useCallback(() => {
+    if (isPrepending.current) {
+      setTopPageCount(prev => prev + 1)
+    }
+  }, [])
```

Comment:
This works well to make sure the scroll animation is completed before prepending. We can't run all of the logic here though, because we don't have any information about the scroll offset. So instead we rely on the ref that gets updated inside of `onStartReached`